### DISCUSSION TITLE
getActivitylogOptions() has to be implemented on every models that uses LogsActivity trait

### DISCUSSION THREAD

**Discussion by acgtwentyone** [spatie/laravel-activitylog#1063] [Ideas]
"The trait contains an abstract method getActivitylogOptions() that you can use to customize options. It needs to return a LogOptions instance built from LogOptions::defaults()" as mentioned on the [docs](https://spatie.be/docs/laravel-activitylog/v4/advanced-usage/logging-model-events) 

Let's say i have thousand of models and just upgrade to from version 3 to version 4.

And then i run ```composer dump-autoload```

The results is something like that:
<img width="1037" alt="Screenshot 2022-07-12 at 11 30 48" src="https://user-images.githubusercontent.com/94224806/178490139-ddd05fa8-b720-4661-9d3d-99d20f27e754.png">

See ? 

In this case i solved this by adding those lines of code:

```php
use Spatie\Activitylog\LogOptions;
```

And then 

```php
public function getActivitylogOptions(): LogOptions
{
    return LogOptions::defaults()
}
```

But with this approach i had to add those line of code on all of my models that were using this ```use LogsActivity``` which may seems to be a better approach.

List of my models (a few of them actually):

<img width="1434" alt="Screenshot 2022-07-12 at 11 21 17" src="https://user-images.githubusercontent.com/94224806/178491748-6f29e434-a74a-4211-bd2a-d610e7cc0989.png">

So i think the ```getActivitylogOptions``` abstract method should be revised.

**Comment by sherifalaa55** [comment]
I know this is an old thread but I just bumped into this and the decision to make us implement the abstract method in each model is really weird... I'd love to know the reasoning for it

**Reply by acgtwentyone** [reply]
@sherifalaa55 It does not makes sense for sure.

**Comment by AshauranceHeavens** [comment]
The documentation on this link cannot be found https://spatie.be/docs/laravel-activitylog/v4/advanced-usage/logging-model-events

**Reply by mlars84** [reply]
Yeah. the docs are also only for v1 or v2. When I installed, it installed v4.7. Seems like they're maintaining the code, but not the documentation. I haven't tried reverting yet, but as long as there are no dependency conflicts, that should follow the docs.

**Reply by AshauranceHeavens** [reply]
Thanks, I was able to get it to work

**Comment by freekmurze** [comment]
Addressed in the v5 branch (#1452). `getActivitylogOptions()` is now optional. You can simply `use LogsActivity` without implementing the method:

```php
class Article extends Model
{
    use LogsActivity;
}
// Logs created/updated/deleted events, but no attribute changes
```

Override it only when you need to customize attribute tracking. Would love feedback on whether the defaults feel right.
